release_process.md

```diff
@@ -1,162 +1,453 @@
-# Release process
+# Release Process
 
-We create a full release of Lance up to every 2 weeks. In between full releases,
-we make preview releases of the latest features and bug fixes, which are hosted
-on fury.io. This allows us to release frequently and get feedback on new features
-while keeping under the PyPI project size limits.
+Lance uses a semantic versioning release workflow with Release Candidates (RC) for voting before stable releases. This ensures quality releases through a structured voting process while maintaining flexibility for urgent patches.
 
-## Overview of Automated Release Process
+## Overview
 
-The Lance release process is now automated using `bump-my-version` to eliminate 
-manual version updates. The workflow handles version bumping, breaking change 
-validation, and release creation automatically.
+The release process follows a structured workflow:
+1. **Development** happens on `main` branch (always at `-beta.N` versions)
+2. **Release Candidates** are created for voting (tagged as `-rc.N`)
+3. **Stable Releases** are promoted from approved RCs
+4. **Patch Releases** are managed through persistent release branches
 
-## Choosing a full versus preview release
+## Branching Strategy
 
-There are three conditions that can trigger a full release:
+### Main Branch
+- Always contains the latest development work
+- Version format: `X.Y.Z-beta.N` (e.g., `1.3.0-beta.2`)
+- Never contains stable or RC versions
+- Automatically bumped after major/minor releases based on breaking changes
 
-1. There's a bugfix we urgently want to get out to a broad audience
-2. We want to make a release of LanceDB that requires new features from Lance
-   (LanceDB can't depend on a preview release of Lance)
-3. It's been two weeks since we last released a full release.
+### Release Branches
+- Format: `release/v{major}.{minor}` (e.g., `release/v1.3`)
+- Created when promoting RC to stable
+- Maintained for patch releases
+- Version format: `X.Y.Z-beta.N` between patch releases
 
-Otherwise, we should make a preview release.
+## Version Flow
 
-## Make a preview release
+```
+main (beta):
+  1.3.0-beta.1 → 1.3.0-beta.2 → ... → 2.0.0-beta.1 → ...
+                       |                      ↑
+                       | (create RC)          | (auto-bump after promote)
+                       ↓                      |
+                [1.3.0-rc.1]            (breaking changes)
+              (detached commit)               |
+                       |                      |
+                       | (promote: derive from original beta commit)
+                       ↓                      |
+release/v1.3:    1.3.0 ────────────────────────
+                   |
+                   ↓
+              1.3.1-beta.1 → ... (for patches)
+
+Note: RC commits (shown in [brackets]) are detached commits tagged for voting.
```

Comment:
Why not have the RC commits live on the release branch? That's where I would expect them to exist.